app/src/main/java/com/example/pokedexdesafio/data/services/GetPokemonDetailService.kt

```diff
@@ -1,33 +1,48 @@
 package com.example.pokedexdesafio.data.services
 
 import android.util.Log
-import com.example.pokedexdesafio.core.utils.ERROR_CODE_IO_EXCEPTION
-import com.example.pokedexdesafio.core.utils.ERROR_CONTENT_LENGTH
-import com.example.pokedexdesafio.core.utils.MEDIA_TYPE_JSON
+import com.example.pokedexdesafio.core.functional.Failure
+import com.example.pokedexdesafio.core.functional.Response
+import com.example.pokedexdesafio.core.functional.Success
+import com.example.pokedexdesafio.core.utils.HTTP_400
+import com.example.pokedexdesafio.core.utils.HTTP_500
 import com.example.pokedexdesafio.data.api.PokemonApi
-import com.example.pokedexdesafio.data.model.PokemonDetail
-import okhttp3.MediaType
-import okhttp3.ResponseBody
-import okio.Buffer
-import retrofit2.Response
+import com.example.pokedexdesafio.data.model.toAbilitiesList
+import com.example.pokedexdesafio.data.model.toMovesList
+import com.example.pokedexdesafio.data.model.toTypesList
+import com.example.pokedexdesafio.domain.model.PokemonDetail
 import retrofit2.Retrofit
 import javax.inject.Inject
 
 class GetPokemonDetailService @Inject constructor(private val retrofit: Retrofit) {
 
     fun fetchPokemonDetails(id: String): Response<PokemonDetail> {
         return try {
-            retrofit.create(PokemonApi::class.java).fetchPokemonDetails(id).execute()
+           val httpResponse = retrofit.create(PokemonApi::class.java).fetchPokemonDetails(id).execute()
+
+            if (httpResponse.isSuccessful) {
+                val listMoves = httpResponse.body()?.toMovesList().orEmpty()
+                val listAbility = httpResponse.body()?.toAbilitiesList().orEmpty()
+                val typeList = httpResponse.body()?.toTypesList().orEmpty()
+                val pokemonDetail = PokemonDetail(
+                    httpResponse.body()?.id,
+                    httpResponse.body()?.name,
```

Comment:
name should be important in the business object. Name must not be a nullable property

Reply:
solved